dask_expr/tests/test_collection.py

```diff
@@ -249,6 +251,18 @@ def test_columns_traverse_filters(pdf, df):
     assert str(result) == str(expected)
 
 
+def test_clip_traverse_filters(df):
+    result = optimize(df.clip(lower=10).y, fuse=False)
+    expected = df.y.clip(lower=10)
+
+    assert str(result) == str(expected)
```

Comment:
I used str early on before we had deterministic hashing.  I think that now the right way to do this is by comparing `._name` (although we'd also welcome a better way if you have thoughts)